docs/dal/index.rst

```diff
@@ -146,10 +146,196 @@ Table Access Protocol
+attribute for some services. For other services like this one, try 
+the ``examples`` attribute.
+
+.. doctest-remote-data::
+
+    >>> print(tap_service.examples[0]['QUERY'])  
+    SELECT TOP 50 l.id, l.pmra as lpmra, l.pmde as lpmde,
+    g.source_id, g.pmra as gpmra, g.pmdec as gpmde
+    FROM
+    lspm.main as l
+    JOIN gaia.dr3lite AS g
+    ON (DISTANCE(g.ra, g.dec, l.raj2000, l.dej2000)<0.01) -- rough pre-selection
+    WHERE
+    DISTANCE(
+        ivo_epoch_prop_pos(
+        g.ra, g.dec, g.parallax,
+        g.pmra, g.pmdec, g.radial_velocity,
+        2016, 2000),
+        POINT(l.raj2000, l.dej2000)
+    )<0.0002                            -- fine selection with PMs
+
+Furthermore, one can find the names of the tables using:
+
+.. doctest-remote-data::
+
+    >>> print([tab_name for tab_name in tap_service.tables.keys()])  # doctest: +ELLIPSIS, +IGNORE_WARNINGS
+    ['amanda.nucand', 'annisred.main', 'antares.data', ..., 'wfpdb.main', 'wise.main', 'zcosmos.data']
+    
+
+And also the names of the columns from a known table, for instance 
+the first three columns:
+
+.. doctest-remote-data::
+
+    >>> result.table.columns[:3]    # doctest: +IGNORE_WARNINGS
+    <TableColumns names=('source_id','ra','dec')>
+
+If you know a TAP service's access URL, you can directly pass it to
+:py:class:`~pyvo.dal.TAPService` to obtain a service object. 
+Sometimes, such URLs are published in papers or passed around through
+other channels. Most commonly, you will discover them in the VO 
+registry (cf. :ref:`pyvo.registry<pyvo-registry>`).
+
+To perform a query using ADQL, the ``search()`` method is used. 
+TAPService instances have several methods to inspect the metadata
+of the service - in particular, what tables with what columns are
+available - discussed below.
+
+To get an idea of how to write queries in ADQL, have a look at
+`GAVO's ADQL course`_; it is basically a standardised subset of SQL
+with some extensions to make it work better for astronomy.
+
+.. _GAVO's ADQL course: https://docs.g-vo.org/adql 
+
+Synchronous vs. asynchronous query
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In synchronous (“sync”) mode, the client keeps a connection for the
+entire runtime of the query, and query processing generally starts
+when the request is submitted.  This is convenient but becomes
+brittle as queries have runtimes of the order of minutes, when you
+may encounter query timeouts.  Also, many data providers impose
+rather strict limits on the runtime alotted to sync queries. 
+
+In asynchronous (“async”) mode, on the other hand, the client just
+submits a query and receives a URL that let us inspect the
+execution status (and retrieve its result) later.  This means that
+no connection needs to be held, which makes this mode a lot more
+robust of long-running queries.  It also supports queuing queries,
+which allows service operators to be a lot more generous with
+resource limits.
```

Comment:
Async queries are not bullet proof either. Service operators might impose limits on the size of the result set which can cause the query to fail or have the result truncated. Also, results might not be available before execution completes (i.e. streaming).